Assets/BossRoom/Scripts/Shared/Infrastructure/ObserverBehaviour.cs

```diff
@@ -0,0 +1,62 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace BossRoom.Scripts.Shared.Infrastructure
+{
+    /// <summary>
+    /// MonoBehaviour that will automatically handle setting up to observe something. It also exposes an event so some other component can effectively observe it as well.
+    /// </summary>
+    public abstract class ObserverBehaviour<T> : MonoBehaviour where T : Observed<T>

```

Comment:
Taking a note here, bigger conversation with SA and Samples needs to happen on this